homeassistant/components/media_player/plex.py

```diff
@@ -7,32 +7,77 @@
 import json
 import logging
 import os
-from datetime import timedelta
+from datetime import datetime, timedelta
 from urllib.parse import urlparse
 
-import homeassistant.util as util
+import requests
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+import voluptuous as vol
+from homeassistant import util
 from homeassistant.components.media_player import (
-    MEDIA_TYPE_TVSHOW, MEDIA_TYPE_VIDEO, MEDIA_TYPE_MUSIC, SUPPORT_NEXT_TRACK,
-    SUPPORT_PREVIOUS_TRACK, SUPPORT_PAUSE, SUPPORT_STOP, SUPPORT_VOLUME_SET,
-    SUPPORT_PLAY, MediaPlayerDevice)
+    MEDIA_TYPE_MUSIC,
+    MEDIA_TYPE_TVSHOW,
+    MEDIA_TYPE_VIDEO,
+    PLATFORM_SCHEMA,
+    SUPPORT_NEXT_TRACK,
+    SUPPORT_PAUSE,
+    SUPPORT_PLAY,
+    SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_SEEK,
+    SUPPORT_STOP,
+    SUPPORT_TURN_OFF,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    MediaPlayerDevice,
+)
 from homeassistant.const import (
-    DEVICE_DEFAULT_NAME, STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING,
-    STATE_UNKNOWN)
+    DEVICE_DEFAULT_NAME,
+    STATE_IDLE,
+    STATE_OFF,
+    STATE_PAUSED,
+    STATE_PLAYING,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import track_utc_time_change
 from homeassistant.loader import get_component
-from homeassistant.helpers.event import (track_utc_time_change)
+
 
 REQUIREMENTS = ['plexapi==2.0.2']
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
 MIN_TIME_BETWEEN_FORCED_SCANS = timedelta(seconds=1)
+DEFAULT_MAX_FROZEN_PLAYING = 60
+DEFAULT_MAX_FROZEN_PAUSED = 300
 
 PLEX_CONFIG_FILE = 'plex.conf'
 
+GROUP_ACTIVE_DEVICES = 'group._plex_devices_active'
+GROUP_INACTIVE_DEVICES = 'group._plex_devices_inactive'
+
+CONF_INCLUDE_NON_CLIENTS = 'include_non_clients'
+CONF_USE_EPISODE_ART = 'use_episode_art'
+CONF_USE_CUSTOM_ENTITY_IDS = 'use_custom_entity_ids'
+CONF_SHOW_ALL_CONTROLS = 'show_all_controls'
+CONF_MAX_FROZEN_PLAYING = 'max_frozen_playing'
+CONF_MAX_FROZEN_PAUSED = 'max_frozen_paused'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_INCLUDE_NON_CLIENTS, default=False):
+    cv.boolean,
+    vol.Optional(CONF_USE_EPISODE_ART, default=False):
+    cv.boolean,
+    vol.Optional(CONF_USE_CUSTOM_ENTITY_IDS, default=False):
+    cv.boolean,
+    vol.Optional(CONF_MAX_FROZEN_PLAYING, default=DEFAULT_MAX_FROZEN_PLAYING):
+    cv.positive_int,
+    vol.Optional(CONF_MAX_FROZEN_PAUSED, default=DEFAULT_MAX_FROZEN_PAUSED):
+    cv.positive_int,
+})
+
 # Map ip to request id for configuring
 _CONFIGURING = {}
 _LOGGER = logging.getLogger(__name__)
 
-SUPPORT_PLEX = SUPPORT_PAUSE | SUPPORT_PREVIOUS_TRACK | SUPPORT_NEXT_TRACK | \
-    SUPPORT_STOP | SUPPORT_VOLUME_SET | SUPPORT_PLAY
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
```

Comment:
So this may be a problem or something we have to think through more. I hit this same problem back when I tried getting PMS to connect via SSL. Full notes are [here](https://github.com/home-assistant/home-assistant/issues/891#issuecomment-275016420), I suggest reading them